packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -381,6 +383,32 @@ export class ApolloServerBase {
     // is populated accordingly.
     this.ensurePluginInstantiation(plugins);
 
+    // We handle signals if it was explicitly requested, or if we're not in a test
+    // and it wasn't explicitly turned off. (For backwards compatibility, we check
+    // both the top-level handleSignals and nested inside 'engine'.)
+    if (
+      handleSignals === true ||
+      (typeof this.config.engine === 'object' &&
+        this.config.engine.handleSignals === true) ||
+      (process.env.NODE_ENV !== 'test' &&
+        handleSignals !== false &&
+        (typeof this.config.engine !== 'object' ||
+          this.config.engine.handleSignals !== false))
+    ) {
```

Comment:
Can you remind me what the best syntax for doing this is?

Reply:
What would be the use case this would not be present?  It seems this should be present due to this being a global object of node, unless there is a name collision within the file.

As for syntax, could do something along the lines of:
`if (process && process.env.NODE_ENV...`
or (event more safe)
`if (process instanceof EventEmitter && process.env.NODE_ENV...`

Reply:
I believe the concern is working towards making as much of Apollo Server as possible not dependent on Node-specific APIs.

Reply:
Ah! Makes sense, thanks!

Reply:
I'm going to go with `isNodeLike` here. (If that's not good enough for your environment you can always specify the option explicitly.)